From 03159e68ef5456b9ea80be813d0af02aac896d2b Mon Sep 17 00:00:00 2001
From: Arulselvan M <arulselvan.m@intel.com>
Date: Fri, 31 May 2013 22:53:50 +0530
Subject: [PORT FROM MAIN] During playback of ALAC5.1 and multichannel wave files Audio is not heard on IHF

BZ: 57613

This patch contains changes in AudioMixer.cpp. Currently mutli-channel SRC
isn't added, so change is done to use stereo when multi-channel downmix
buffer is present.

Change-Id: Ia002e4240211e690bffb82d6de4341bcd9a3117a
Orig-Change-Id: If23e1bec0f340f0d4e965fcdd071d38a181e8860
Signed-off-by: Manjunath <manjunathx.b.s@intel.com>
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 services/audioflinger/AudioMixer.cpp |   14 ++++++++------
 1 files changed, 8 insertions(+), 6 deletions(-)

diff --git a/services/audioflinger/AudioMixer.cpp b/services/audioflinger/AudioMixer.cpp
index 951266e..b18d90e 100644
--- a/services/audioflinger/AudioMixer.cpp
+++ b/services/audioflinger/AudioMixer.cpp
@@ -553,13 +553,16 @@ bool AudioMixer::track_t::setResampler(uint32_t value, uint32_t devSampleRate)
     if (value != devSampleRate || resampler != NULL) {
         if (sampleRate != value) {
             sampleRate = value;
+
+            uint8_t channels = downmixerBufferProvider != NULL ? MAX_NUM_CHANNELS : channelCount;
+
             if (resampler == NULL) {
 #ifdef USE_INTEL_SRC
                 if (AudioResamplerIA::sampleRateSupported(sampleRate, devSampleRate)) {
-                    resampler = AudioResampler::create(format, channelCount,
+                    resampler = AudioResampler::create(format, channels,
                                 devSampleRate, AudioResampler::INTEL_HIGH_QUALITY);
                 } else {
-                    resampler = AudioResampler::create(format, channelCount, devSampleRate);
+                    resampler = AudioResampler::create(format, channels, devSampleRate);
                 }
 #else
                 ALOGV("creating resampler from track %d Hz to device %d Hz", value, devSampleRate);
@@ -575,9 +578,7 @@ bool AudioMixer::track_t::setResampler(uint32_t value, uint32_t devSampleRate)
                     quality = AudioResampler::DEFAULT_QUALITY;
                 }
                 resampler = AudioResampler::create(
-                        format,
-                        // the resampler sees the number of channels after the downmixer, if any
-                        downmixerBufferProvider != NULL ? MAX_NUM_CHANNELS : channelCount,
+                        format, channels,
                         devSampleRate, quality);
                 resampler->setLocalTimeFreq(sLocalTimeFreq);
 #endif
@@ -587,7 +588,8 @@ bool AudioMixer::track_t::setResampler(uint32_t value, uint32_t devSampleRate)
                 if (resampler->mResType == AudioResampler::INTEL_SRC &&
                     !AudioResamplerIA::sampleRateSupported(sampleRate, devSampleRate)) {
                     delete resampler;
-                    resampler = AudioResampler::create(format, channelCount, devSampleRate);
+                    resampler = AudioResampler::create(format, channels, devSampleRate);
+                    resampler->setLocalTimeFreq(sLocalTimeFreq);
                 }
             }
 #endif
-- 
1.7.4.1

